modules/op-interacting-with-pipelines-using-the-developer-perspective.adoc

```diff
@@ -16,28 +16,38 @@ The *Pipelines* view in the *Developer* perspective lists all the pipelines in a
 [Discrete]
 .Procedure
 . In the *Pipelines* view of the *Developer* perspective, select a project from the *Project* drop-down list to see the pipelines in that project.
-. Click the required pipeline to see the *Pipeline details* page. By default, the *Details* tab opens and provides a visual representation of all the serial and parallel tasks in the pipeline. The tasks are also listed in the lower right portion of the page. You can click the listed *Tasks* to view the task details.
+. Click the required pipeline to see the *Pipeline details* page. By default, the *Details* tab opens and provides a visual representation of all the serial tasks, parallel tasks, and the final tasks, along with the `when` expressions, if any, in the pipeline. The tasks and the final tasks are also listed in the lower right portion of the page. You can click the listed *Tasks* and *Finally tasks* to view the task details.
 +
 .Pipeline details
 image::op-pipeline-details.png[Pipeline details]
 +
-. Optionally, in the *Pipeline details* page:
-* Click the *Metrics* tab to see the following information about pipelines:
+. Optional: In the *Pipeline details* page, click the *Metrics* tab to see the following information about pipelines:
 ** *Pipeline Success Ratio*
 ** *Number of Pipeline Runs*
 ** *Pipeline Run Duration*
 ** *Task Run Duration*
 +
 You can use this information to improve the pipeline workflow and eliminate issues early in the pipeline lifecycle.
 +
-* Click the *YAML* tab to edit the YAML file for the pipeline.
-* Click the *Pipeline Runs* tab to see the completed, running, or failed runs for the pipeline.
+. Optional: Click the *YAML* tab to edit the YAML file for the pipeline.
+. Optional: Click the *Pipeline Runs* tab to see the completed, running, or failed runs for the pipeline.
++
+The *Pipeline Runs* tab provides information about the pipeline run and its details, the status of the task, and a link to debug failed pipeline runs. You can use the Options menu {kebab} to stop a running pipeline, to rerun a pipeline using the same parameters and resources as that of the previous pipeline execution, or to delete a pipeline run.
++
+* Click the required pipeline run to see the *Pipeline Run details* page. By default, the *Details* tab opens and provides a visual representation of all the serial tasks, parallel tasks, and the final tasks, along with the `when` expressions, if any, in the PipelineRun. Only if a pipeline run is successful, the results are displayed under the *Pipeline Run results* pane at the bottom of the page.
```

Comment:
when expressions, if any, in the pipeline run.

Reply:
"By default, the ... displays..." 
And, not along with. 
Remove if any. That is understood. 
The results are displayed when the run is successful.